octopus-agent/src/main/java/octopus/teamcity/agent/logging/SdkLogAppender.java

```diff
@@ -0,0 +1,73 @@
+package octopus.teamcity.agent.logging;
+
+import java.io.Serializable;
+import java.util.Map;
+
+import jetbrains.buildServer.agent.BuildProgressLogger;
+import octopus.teamcity.common.commonstep.CommonStepPropertyNames;
+import org.apache.logging.log4j.Level;
+import org.apache.logging.log4j.core.Appender;
+import org.apache.logging.log4j.core.Core;
+import org.apache.logging.log4j.core.Filter;
+import org.apache.logging.log4j.core.Layout;
+import org.apache.logging.log4j.core.LogEvent;
+import org.apache.logging.log4j.core.appender.AbstractAppender;
+import org.apache.logging.log4j.core.config.Property;
+import org.apache.logging.log4j.core.config.plugins.Plugin;
+import org.apache.logging.log4j.core.config.plugins.PluginAttribute;
+import org.apache.logging.log4j.core.config.plugins.PluginElement;
+import org.apache.logging.log4j.core.config.plugins.PluginFactory;
+import org.apache.logging.log4j.core.layout.PatternLayout;
+
+@Plugin(
+    name = SdkLogAppender.SDK_APPENDER_NAME,
+    category = Core.CATEGORY_NAME,
+    elementType = Appender.ELEMENT_TYPE,
+    printObject = true)
+public class SdkLogAppender extends AbstractAppender {
+
+  public static final String SDK_APPENDER_NAME = "SdkLogAppender";
+
+  private final BuildProgressLogger buildProgressLogger;
+
+  protected SdkLogAppender(
```

Comment:
final on parameters (you should be able to update your templates to do this for you.